docs/blueprints/storage-node-downtime-disqualification.md

```diff
@@ -0,0 +1,107 @@
+
+Scale the _weight_ of the uptime check based on time elapsed since last contact.
+
+The previous uptime reputation system worked much like the current audit reputation. See [Node Selection](node-selection.md) for more information.
+> α(n) = λ·α(n-1) + _w_(1+_v_)/2
+>
+> β(n) = λ·β(n-1) + _w_(1-_v_)/2
+>
+> R(n) = α(n) / (α(n) + β(n))
+
+In the case of a successful uptime check, _v_ is set to 1. Otherwise, it is set to -1. Previously, _w_, or _weight_, was configured at runtime and thereafter remained static. Thus, all uptime checks carried the same weight. However, should this really be the case?
+For example, a node fails two uptime checks within a timespan of 1 minute. The second failure does not give us very much additional information does it? We already knew the node was offline less than one minute ago. If the difference were one hour that information might be more useful in determining uptime.
+
+We might be able to account for this by adjusting _w_ based on how much time has elapsed since last contact.
+To do this we'll need a new configurable `Uptime Scale`: a duration against which to scale the weight of an individual uptime check. To demonstrate, let's take an Uptime Scale of one hour. If a node is found to be offline when audited and its last contact is greater than or equal to one hour ago, the weight is scaled to 100%. This check will result in the full reputation punishment. A second failed check occurs 30 minutes later. This is one half the scale: the weight is reduced by 50%. Another failed check occurs one minute later, and the weight is scaled back to 1/60 the punishment.
+
+### Tracking period
+One might think to measure downtime within a static window, such as within month boundaries, after which downtime is reset. This introduces a problem. If one day of downtime within the tracking period is allowed, why _not_ shut down on the final day? This is compounded by the fact that any number of nodes could have the same idea, resulting in a potentially significant portion the network going offline at the same time. The trailing tracking period does not fall victim to the same problem. There is no specific timeframe which has a higher incentive for downtime, and any observed downtime will follow the node for the length of the tracking period.
+
+### Early reinstatement for good nodes?
+An alternate approach to suspending for a required length of time before reinstatement could be that we periodically evaluate a node's downtime during suspension. If the node's downtime has fallen within the acceptable range, we reinstate it. However, this could result in nodes falling in and out of suspension. If we suspend the node for a fixed period of time and evaluate its downtime at the end of the period, we can limit the rate of fluctuation.
```

Comment:
I would prefer to risk the fluctuation because I don't think over a periode of 30 days many nodes will hover around the magic line. I would expect that nodes that are able to exit suspension mode early hopefully stay healty for a while. I would appreciate that kind of fluctuation. We only have to step in if they enter and exit suspension mode frequently. Lets find out if that will happen before we punish all nodes for that. The downside of having no early reinstatement is that we would be forced to keep the graceful periode short. The longer the periode is the longer we generate some additional repair traffic in our system. Early reinstatement would help us to reduce the repair cost.